src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpImportServiceDAO.scala

```diff
@@ -21,10 +21,17 @@ class HttpImportServiceDAO(implicit val system: ActorSystem, implicit val materi
   extends ImportServiceDAO with RestJsonClient with SprayJsonSupport {
 
   override def importPFB(workspaceNamespace: String, workspaceName: String, pfbRequest: PfbImportRequest)(implicit userInfo: UserInfo): Future[PerRequestMessage] = {
+    doImport(workspaceNamespace, workspaceName, pfbRequest, "pfb")
+  }
+
+  override def importBatchUpsertJson(workspaceNamespace: String, workspaceName: String, pfbRequest: PfbImportRequest)(implicit userInfo: UserInfo): Future[PerRequestMessage] = {
+    doImport(workspaceNamespace, workspaceName, pfbRequest, "rawlsjson")
+  }
 
+  private def doImport(workspaceNamespace: String, workspaceName: String, pfbRequest: PfbImportRequest, filetype: String)(implicit userInfo: UserInfo): Future[PerRequestMessage] = {
     // the payload to Import Service sends "path" and filetype.  Here, we force-hardcode filetype because this API
     // should only be used for PFBs.
-    val importServicePayload: ImportServiceRequest = ImportServiceRequest(path = pfbRequest.url.getOrElse(""), filetype = "pfb")
+    val importServicePayload: ImportServiceRequest = ImportServiceRequest(path = pfbRequest.url.getOrElse(""), filetype = filetype)
```

Comment:
Could we throw an exception if pfbRequest.url is null?  Setting it to empty string will cause an error further from the calling code in import service I believe?

Reply:
Another good catch. Previously (https://github.com/broadinstitute/firecloud-orchestration/pull/777/files#r395754726) I'd been passing on potentially-invalid payloads to Import Service, and letting Import Service do all the validation and produce all the error messages. But now that we're handling both TSVs (which orch mostly owns) and PFBs (which Import Service owns) I agree about validating here. I've updated the code.